include/nodes/internal/Connection.hpp

```diff
@@ -139,10 +145,11 @@ class NODE_EDITOR_PUBLIC Connection
   PortIndex _inPortIndex;
 
 private:
-
   ConnectionState    _connectionState;
   ConnectionGeometry _connectionGeometry;
 
+  ConnectionStyle const *_connectionStyle;
```

Comment:
Sorry, maybe a stupid question -- why pointer?

Reply:
What other options were you thinking of?

* `ConnectionStyle _connectionStyle;` This is expensive. We don't need a copy of the `ConnectionStyle` object for each `Connection`, especially when they're all going to be the same
* `std::shared_ptr<ConnectionStyle const> _connectionStyle;` This is unnecessary. The `ConnectionStyle` outlives the connections anyway (correction: they can, but currently don't; that's something I will fix).
* `ConnectionStyle const &_connectionStyle;` I don't use references as class members, as references can't be rebound (they are effectively const). It doesn't matter here, as `Connection` can't be copied anyway, but I used a pointer instead for consistency.

Reply:
Here `ConnectionStyle const & connectionStyle() const` we convert this pointer back to a reference.

Reply:
True. The member is stored as a pointer, but is semantically a reference (that's the style I use anyway).

Reply:
1. I did not see any setter function
2. The copy constructor for `Connection` is deleted.

Therefore I suggested that this member shouldn't be rebound.

I'd simply use a const ref.

_BTW, we should explicitly delete move constructor and move assignment operator for this class_